From 16870ba643ed8c45882d5707bbd9a477c83c5d7e Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Mon, 08 Jul 2024 17:47:57 +0300
Subject: [PATCH] [FSW-420] fs1: check signature val efuse bit

witht this patch, added support to check
signature validation efuse syscfg bit, this
efuse bit check required to check before
all image validation api's.

Change-Id: I21bf311a189ed636f34c0494488110320c348b7e
---

diff --git a/arch/fs1/arc/dev_security.c b/arch/fs1/arc/dev_security.c
index 31d6b3c..9c854ff 100644
--- a/arch/fs1/arc/dev_security.c
+++ b/arch/fs1/arc/dev_security.c
@@ -84,3 +84,11 @@
 
 	boot_rom_global.sec_data.key_revoke = key_revoke;
 }
+
+int security_efuse_image_validation_en(void)
+{
+	/* check efuse for security validation enable */
+	uint32_t sys_cfg = boot_rom_global.sec_data.efuse_sys_cfg;
+
+	return !!(sys_cfg & EFUSE_SYSCFG_SIG_VAL_EN);
+}
diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
index 4d84965..68e7352 100644
--- a/arch/fs1/arc/dev_security.h
+++ b/arch/fs1/arc/dev_security.h
@@ -9,6 +9,7 @@
 #define NUM_KEYS_IN_ROM					5
 
 void security_load_revoke_bits(void);
+int security_efuse_image_validation_en(void);
 
 extern const uint8_t rom_key_hashes[NUM_KEYS_IN_ROM][SHA384_SUM_LEN];
 
